File: CBMCalculator/app/src/main/java/com/cbm/android/cbmcalculator/tool/Tools.java
package com.cbm.android.cbmcalculator.tool;

import java.math.BigDecimal;

public class Tools {
    public static boolean isNumber(String text) {
        try {
            BigDecimal bd = new BigDecimal(text);
            return !(bd.doubleValue()+"").isEmpty();
        } catch (Exception e){e.printStackTrace();}
        return false;
    }
}

File: CBMCalculatorAndroid/app/src/main/java/com/cbm/android/cbmcalculatorandroid/Tools.kt
package com.cbm.android.cbmcalculatorandroid

import java.math.BigDecimal

class Tools {
    companion object {
        public fun isNumber(value:String): Boolean {
            try {
                val db = BigDecimal(value);
                return !db.toDouble().toString().isEmpty()
            } catch(ex:Exception){return false}
        }

        public fun expression():ArrayList<String> {
            val arr = ArrayList<String>()
            arr.add("+")
            arr.add("-")
            arr.add("")
            arr.add("")
            return arr
        }
     }
}